### PR TITLE
Adapt Gyroscope to the latest changes in the Generic Sensor API

### PR DESCRIPTION
**Pull request by pozdnyakov**
This patch gets rids of 'Reading' class and re-defines reading attributes using "latest reading" terminology from Generic Sensor API (https://w3c.github.io/sensors/#latest-reading).

### CODE REVIEW

**Comment by pozdnyakov** [comment]
@anssiko PTAL

File: index.bs [hunk 1]
```diff
@@ -75,29 +75,18 @@ Model {#model}
...
 
-The Gyroscope has an associated abstract operation
-to <dfn lt="Construct SensorReading Object">construct a SensorReading object</dfn>
-which creates a new {{GyroscopeReading}} object and sets each of its
-<a attribute for="GyroscopeReading">x</a>
-<a attribute for="GyroscopeReading">y</a> and
-<a attribute for="GyroscopeReading">z</a> attributes to
-the <a>current angular velocity</a> about the corresponding axis.
+The Gyroscope extends sensor [=latest reading=] with three [=map/entries=] whose [=map/keys=] are "x", "y", "z" and whose [=map/values=]
+contain <a>current angular velocity</a> about the corresponding axes.
```
Comment (alexshalamov) [inline index.bs]: about => around?

**Comment by anssiko** [comment]
Proposed to be added to the agenda for today's call.

File: index.bs [hunk 2]
```diff
@@ -16,7 +16,7 @@ Version History: https://github.com/w3c/gyroscope/commits/gh-pages/index.bs
 Indent: 2
 Repository: w3c/gyroscope
 Markup Shorthands: markdown on
-Inline Github Issues: true
+Inline Github Issues: title
 !Test Suite: <a href="https://github.com/w3c/web-platform-tests/tree/master/gyroscope">web-platform-tests on GitHub</a>
 Boilerplate: omit issues-index, omit conformance
 Default Biblio Status: current
```
Comment (anssiko) [inline index.bs:22]: Doing a link check, noticed the biblio fragments on L27-28 are broken. The Permissions spec has changed from underneath us. Upstream PR: https://github.com/w3c/sensors/pull/157

File: index.bs [hunk 3]
```diff
@@ -30,7 +30,7 @@ urlPrefix: https://w3c.github.io/sensors; spec: GENERIC-SENSOR
   type: dfn
     text: high-level
     text: sensor subclass
```
Comment (anssiko) [inline index.bs]: `text: sensor`

File: index.bs [hunk 4]
```diff
@@ -75,29 +75,18 @@ Model {#model}
 The Gyroscope's associated <a>Sensor subclass</a>
```
Comment (anssiko) [inline index.bs]: `<a>Sensor</a> subclass`

File: index.bs [hunk 5]
```diff
@@ -75,29 +75,18 @@ Model {#model}
...
 must simply return a <a>permission</a> whose name is "gyroscope".
 
-The Gyroscope has an associated abstract operation
-to <dfn lt="Construct SensorReading Object">construct a SensorReading object</dfn>
-which creates a new {{GyroscopeReading}} object and sets each of its
-<a attribute for="GyroscopeReading">x</a>
-<a attribute for="GyroscopeReading">y</a> and
-<a attribute for="GyroscopeReading">z</a> attributes to
-the <a>current angular velocity</a> about the corresponding axis.
+The Gyroscope extends sensor [=latest reading=] with three [=map/entries=] whose [=map/keys=] are "x", "y", "z" and whose [=map/values=]
```
Comment (anssiko) [inline index.bs]: `extends sensor's`

**Comment by anssiko** [comment]
@tobie just committed to review this PR on the WG call.

File: index.bs (hunk 2 above)
Comment (anssiko) [reply index.bs:22]: We don't need to block on this, let's fix these when the upstream PR lands.

Comment (anssiko) [reply index.bs:22]: @pozdnyakov https://github.com/w3c/sensors/pull/157 landed so you may want to sync accordingly.

Comment (pozdnyakov) [reply index.bs:22]: fixed by removal

File: index.bs [hunk 6]
```diff
@@ -16,21 +16,17 @@ Version History: https://github.com/w3c/gyroscope/commits/gh-pages/index.bs
...
 urlPrefix: https://w3c.github.io/sensors; spec: GENERIC-SENSOR
   type: dfn
     text: high-level
-    text: sensor subclass
-    text: sensorreading subclass
+    text: sensor
+    text: latest reading
     text: default sensor
     text: supported reporting mode; url: supported-reporting-modes
     text: auto
```
Comment (tobie) [inline index.bs]: You can also get rid of "auto" and "supported reporting mode" here.

File: index.bs [hunk 7]
```diff
@@ -72,32 +68,15 @@ beyond those described in the Generic Sensor API [[!GENERIC-SENSOR]].
 Model {#model}
 =====
 
-The Gyroscope's associated <a>Sensor subclass</a>
+The Gyroscope's associated <a>Sensor</a> subclass
```
Comment (tobie) [inline index.bs]: This should be {{Sensor}}.

File: index.bs [hunk 8]
```diff
@@ -72,32 +68,15 @@ beyond those described in the Generic Sensor API [[!GENERIC-SENSOR]].
...
-must simply return a <a>permission</a> whose name is "gyroscope".
-
-The Gyroscope has an associated abstract operation
-to <dfn lt="Construct SensorReading Object">construct a SensorReading object</dfn>
-which creates a new {{GyroscopeReading}} object and sets each of its
-<a attribute for="GyroscopeReading">x</a>
-<a attribute for="GyroscopeReading">y</a> and
-<a attribute for="GyroscopeReading">z</a> attributes to
-the <a>current angular velocity</a> about the corresponding axis.
+The Gyroscope extends sensor's [=latest reading=] with three [=map/entries=] whose
```
Comment (tobie) [inline index.bs]: It's unclear what "The Gyroscope" refers to, here. Also why the uppercase?

I also would not use the term "extend" here.

There should be a [=latest reading=] per [=sensor=] of type gyroscope.

File: index.bs [hunk 9]
```diff
@@ -118,43 +97,28 @@ The Gyroscope Interface {#gyroscope-interface}
...
-  };
-</pre>
-
-### The GyroscopeReading constructor ### {#gyroscope-reading-constructor}
-
-### The attribute ### {#gyroscope-reading-attribute}
+### The attribute ### {#gyroscope-attribute}
 
-The <a attribute for="GyroscopeReading">x</a> attribute of the {{GyroscopeReading}}
+The <a attribute for="Gyroscope">x</a> attribute of the {{Gyroscope}}
```
Comment (tobie) [inline index.bs]: FYI: can write this as {{Gyroscope/x!!attribute}}. Your choice, however.

**Comment by pozdnyakov** [comment]
@tobie, thanks for your comments! Addressed in the latest patch.

File: index.bs [hunk 10]
```diff
@@ -118,43 +95,28 @@ The Gyroscope Interface {#gyroscope-interface}
...
-      unrestricted double x = 0;
-      unrestricted double y = 0;
-      unrestricted double z = 0;
-  };
-</pre>
-
-### The GyroscopeReading constructor ### {#gyroscope-reading-constructor}
-
-### The attribute ### {#gyroscope-reading-attribute}
+### The attribute ### {#gyroscope-attribute}
```
Comment (tobie) [inline index.bs]: I'd title that differently. See how it's generally done in other specs: one title per attribute.

File: index.bs [hunk 11]
```diff
@@ -72,32 +66,15 @@ beyond those described in the Generic Sensor API [[!GENERIC-SENSOR]].
 Model {#model}
 =====
 
-The Gyroscope's associated <a>Sensor subclass</a>
+The Gyroscope's associated {{Sensor}} subclass
```
Comment (tobie) [inline index.bs]: Same capitalization issue here (and throughout the spec). Would be good to link these, too. Overall, it seems the model on the generic sensor side isn't clear enough. I need to work on it.

**Comment by pozdnyakov** [comment]
@tobie, could you please take another look?

**Review by tobie — APPROVED** [review]

**Pull request merged**